datatypes/src/collections/ipc.rs

```diff
@@ -0,0 +1,191 @@
+use std::{collections::HashMap, sync::Arc};
+
+use super::{FeatureCollection, IntoGeometryOptionsIterator};
+use crate::{spatial_reference::SpatialReferenceOption, util::Result};
+use arrow::{
+    array::{Array, ArrayRef},
+    datatypes::{Field, Schema},
+    ipc::writer::{FileWriter, IpcWriteOptions},
+    record_batch::RecordBatch,
+};
+use wkt::ToWkt;
+
+pub const SPATIAL_REF_KEY: &str = "spatialReference";
+
+pub trait FeatureCollectionIpc<'g> {
+    fn to_arrow_record_batch(&'g self) -> Result<arrow::record_batch::RecordBatch>;
+
+    fn to_arrow_ipc_file_bytes(&'g self, spatial_ref: SpatialReferenceOption) -> Result<Vec<u8>>;
+}
+
+impl<'g, G> FeatureCollectionIpc<'g> for FeatureCollection<G>
+where
+    Self: IntoGeometryOptionsIterator<'g>,
+{
+    fn to_arrow_record_batch(&'g self) -> Result<arrow::record_batch::RecordBatch> {
+        // Currently, PyArrow does not support our storage of geometries
+        // Thus, we will transform them to WKTs.
+        // TODO: use `let record_batch: RecordBatch = struct_ref.into();` instead
```

Comment:
can you add some information when to switch that?

Reply:
I don't know, I didn't find any information on that. It just did not work.